### PR TITLE
Async RNG seed upload with env var `XLA_TRANSFER_SEED_ASYNC=1` (#3292)

### PR DESCRIPTION
**Pull request by JackCaoG**
Back-porting https://github.com/pytorch/xla/pull/3292 to 1.11

### CODE REVIEW

File: third_party/xla_client/computation_client.h [hunk 1]
```diff
@@ -166,10 +167,25 @@ class ComputationClient {
   // will be populated in an asynchrounous fashion.
   virtual DataPtr CreateDataPlaceholder(std::string device, Shape shape) = 0;
 
+  // Create DataPtr that only has dummy information which can be filled in
+  // later.
+  virtual std::vector<DataPtr> CreateAsyncDatas(
```
Comment (yeounoh) [inline third_party/xla_client/computation_client.h:172]: nit. `Datas` to `Data` ?

Comment (JackCaoG) [reply third_party/xla_client/computation_client.h:172]: hmm, it takes vector or tensors and return vecotr of DataPtrs. Datas seems fine? wdyt?

Comment (JackCaoG) [reply third_party/xla_client/computation_client.h:172]: I should update `Create DataPtr` to `Create DataPtrs` through.. I can do that on master instead

**Review by yeounoh — APPROVED** [review]
LGTM

File: third_party/xla_client/computation_client.h (hunk 1 above)
Comment (yeounoh) [reply third_party/xla_client/computation_client.h:172]: Sounds good, `DataPtrs` would make sense to describe the vector, as data is already plural :p Let's do naming changes later in master. Thank you @JackCaoG

**Pull request merged**